packages/main/src/SplitButton.ts

```diff
@@ -300,17 +299,24 @@ class SplitButton extends UI5Element {
 		}
 		this._shiftOrEscapePressed = false;
 		this.focused = false;
-		this._setTabIndexValue();
 	}
 
 	_onFocusIn(e: FocusEvent) {
 		if (this.disabled || getEventMark(e)) {
 			return;
 		}
+		if (!(e.target! as HTMLElement).classList.contains("ui5-split-button-root")) {
+			return;
+		}
 		this._shiftOrEscapePressed = false;
 		this.focused = true;
 	}
 
+	_singleButtonFocusIn(e: FocusEvent) {
```

Comment:
We could react in the `_onFocusIn` Instead of reacting on focus in for the text and arrow buttons separately.
That way we could remove that attached handlers for focusin and focusout for the arrow and text buttons.